projects/wave-core-new/src/lib/operators/dialogs/helpers/multi-layer-selection/multi-layer-selection.component.spec.ts

```diff
@@ -0,0 +1,208 @@
+            imports: [...MATERIAL_MODULES, NoopAnimationsModule],
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(MultiLayerSelectionComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    /** create app **/
+    it('should create the app', () => {
+        expect(component).toBeTruthy();
+    });
+
+    /** initially no Layers -> 'No Input Available' should be displayed **/
+    it('should display "No Input Available", if no Layer is available', () => {
+        html = fixture.nativeElement.querySelector('p');
+        expect(component.selectedLayers.value).toEqual([]);
+        expect(html.textContent).toEqual('No Input Available');
+    });
+
+    /** Adding three layers
+     * checking the number of possible layers to select from
+     * checking the default layer displayed **/
+    it('testing with min = max = 1', async () => {
+        component.ngOnChanges({layers: new SimpleChange(undefined, component.layers, true)});
+        fixture.detectChanges();
+        html = fixture.nativeElement.querySelector('mat-select');
+        html.click();
+        fixture.detectChanges();
+        await fixture.whenStable().then(() => {
+            const inquiryOptions = fixture.debugElement.queryAll(By.css('.mat-option-0'));
+            expect(inquiryOptions.length).toEqual(mockLayers.length);
+            expect(component.selectedLayers.value).not.toEqual([]);
+            expect(html.textContent).toEqual(component.selectedLayers.value[0].name);
+        });
+    });
+
+    /** checking the layer displayed after changing the selected layer **/
+    it('should update the layer displayed to equal selected layer, min = max = 1', async () => {
+        component.ngOnChanges({layers: new SimpleChange(undefined, component.layers, true)});
+        fixture.detectChanges();
+        for (let i = mockLayers.length; i > 0; i--) {
+            component.updateLayer(0, mockLayers[i - 1]);
+            fixture.detectChanges();
+            html = fixture.nativeElement.querySelector('mat-select');
+            html.click();
+            fixture.detectChanges();
+            await fixture.whenStable().then(() => {
+                expect(component.selectedLayers.value).not.toEqual([]);
+                expect(html.textContent).toEqual(component.selectedLayers.value[0].name);
+                expect(html.textContent).toEqual(mockLayers[i - 1].name);
+            });
+        }
+    });
+
+    /** Adding three layers
+     * adding two more input fields
+     * checking the default layers displayed and the number of possible layers to select **/
+    it('testing with max = 3', async () => {
+        component.ngOnChanges({max: new SimpleChange(component.max, 3, true), layers: new SimpleChange(undefined, component.layers, true)});
+        fixture.detectChanges();
+        component.add();
+        component.add();
+        const amount = component.selectedLayers.value.length;
```

Comment:
can we assert the number?

Reply:
eb7fda1155207b8594d05d8c3a338321cca92556